source/grid/grid_tools_cache.cc

```diff
@@ -160,42 +188,31 @@ namespace GridTools
 
   template <int dim, int spacedim>
   const RTree<std::pair<BoundingBox<spacedim>, unsigned int>> &
-  Cache<dim, spacedim>::get_covering_rtree() const
+  Cache<dim, spacedim>::get_covering_rtree(const unsigned int level) const
   {
-    if (update_flags & update_covering_rtree)
+    if (update_flags & update_covering_rtree ||
+        covering_rtree.find(level) == covering_rtree.end())
```

Comment:
Hm, I think I'm starting to realize that "level" has nothing to do with the level of a triangulation, but has some other meaning in rtrees. I think it would be useful to clarify this in pretty much all places of the documentation...

Reply:
the function we used is called `extract_rtree_level`. Should we change the name of that function? Level refers to that. I'll try to reformulate the documentation accordingly.